src/Nancy/IO/RequestStream.cs

```diff
@@ -358,9 +358,28 @@ private void ShiftStreamToFileStreamIfNecessary()
 
         private static FileStream CreateTemporaryFileStream()
         {
-            var filePath = Path.GetTempFileName();
+            try
+            {
+                var filePath = Path.GetTempFileName();
+
+                return new FileStream(
+                    filePath,
+                    FileMode.Create,
+                    FileAccess.ReadWrite,
+                    FileShare.None,
+                    8192,
+                    StaticConfiguration.AllowFileStreamUploadAsync);
+            }
+            catch (System.IO.IOException ex)
+            {
+                if (ex.Message == "The file exists.")

```

Comment:
Probably break this out to a const and also use a case-insensitive string comparison